### ISSUE TITLE
Semantic: forward references can resolve to container fields, causing `unused-decls` false positives

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#366]
# Semantic: forward references can resolve to container *fields*, causing `unused-decls` false positives

- **Severity:** High (produces a default-on false warning, and its dangerous autofix deletes a *used* declaration)
- **Status:** Verified by reproduction
- **Location:** `src/Semantic/Builder.zig:1599-1623` (deferred resolution) vs `src/Semantic/Builder.zig:837-841` (eager resolution)

## The bug

Eager identifier resolution correctly excludes container members:

```zig
// Builder.zig:837
const symbol = self._semantic.resolveBinding(
    self.currentScope(),
    identifier_name,
    .{ .exclude = .{ .s_member = true } },
);
```

But deferred resolution — which runs when a scope exits and tries to resolve references that couldn't be resolved eagerly (forward references) — matches **by name only**, with no flags check:

```zig
// Builder.zig:1615
if (mem.eql(u8, name, ref_name)) {
    num_resolved += 1;
    resolved_map[i] = true;
    ref_symbols[ref] = binding.into(Symbol.Id.Optional);
    try symbol_refs[binding.int()].append(self._gpa, ref_id);
}
```

Container fields are bound into the container's scope, so a forward reference bubbling out of a method gets captured by a same-named *field* instead of the real declaration further out.

## User impact

Valid, idiomatic Zig produces a false `unused-decls` warning (on by default), and the rule's autofix — reachable via plain `--fix` per report 04 — **deletes a declaration that is actually used**, breaking the build.

## Reproduction

```sh
cat > sem1.zig <<'EOF'
const Inner = struct {
    limit: u32,               // field with the same name
    fn f() u32 {
        return limit;         // real Zig resolves this to the root `limit`
    }
};
const limit: u32 = 10;        // declared after Inner -> forward reference
pub fn main() void {
    _ = Inner.f();
    _ = Inner{ .limit = 1 };
}
EOF
zlint sem1.zig
# warning: unused-decls: variable 'limit' is declared but never used.
```

The reference to `limit` in `f` fails eager resolution (root `limit` not yet declared; the field is excluded), bubbles up, and when `Inner`'s container scope exits, the name-only match binds it to the field. The root `const limit` ends up with zero references.

## Proposed fix

Apply the same member exclusion in the deferred path. When iterating a scope's bindings during scope-exit resolution, skip symbols with `flags.s_member` (fields cannot be referred to by bare identifier from nested function scopes in Zig). Concretely, in the resolution loop around `Builder.zig:1599`, check the binding's symbol flags before the name comparison:

```zig
const flags = self.symbolTable().symbols.items(.flags);
for (bindings) |binding| {
    if (flags[binding.int()].s_member) continue;
    ...
}
```

Add a regression test with the snippet above asserting `limit` (root) has one reference and the field has none.
